doc/developer/reference/compute/active-replication.md

```diff
@@ -0,0 +1,46 @@
+
+Given the reliance on other components for persistent state, the mechanism is currently not coupled with a checkpointing facility. Checkpointing by itself does not address the [cost of downtime](https://www.usenix.org/legacy/events/lisa2002/tech/full_papers/patterson/patterson.pdf), which tends to be significant in applications relying on low-latency processing of data streams. When additionally considering the concerns of economy of mechanism for the [intended goals](active-replication.md#active-replication) and manageable complexity, active replication without checkpointing becomes an attractive choice.
+
+### History Reduction
+
+Over time, the compute command history of an instance can grow large. To avoid unbounded growth, the history is reduced, following a length-doubling policy, to an equivalent, hopefully shorter history. The defining property of a reduced history is that its full application by a new replica would bring the replica to the same state as other replicas that have fully applied the unreduced version of the instance's history. More details about history equivalence and reduction are discussed in the [Compute Command Model](compute-command-model.md).
+
+### Failure Detection and Rehydration
+
+Since durability is provided by the ADAPTER and STORAGE layers, the mechanism is intended to tolerate crashes of any number of replicas in an instance. However, masking of recovery delay can only be guaranteed when the compute controller can reach at least one non-faulty replica.
+
+The compute controller assumes detection of crashes by breakdown of the network communication with a replica, namely when sending commands or receiving responses from the replica. Upon such an assumed detection of a crash, the compute controller adds the replica to the instance's set of presumably failed replicas. The latter makes the replica a target for rehydration, i.e., the process of removing the old replica metadata, re-adding the replica, and sending to the replica the entire reduced history of the instance for replay. Note that when re-adding the replica here, an attempt is made to find an existing incarnation of the replica's service, in which case the [Reconciliation](reconciliation.md) optimization can be triggered; if such an incarnation cannot be found, a new replica service is spawned.
+
+Note that a failure of the compute controller will result in unavailability and — at present — loss of the authoritative compute command history. However, the ADAPTER layer can execute a recovery process from the information recorded in the catalog that will endow the compute controller with a history that is equivalent to the history prior to the crash. Additionally, all replicas will be re-added to the compute controller and undergo rehydration.
+
+## Core Components in Source Code
+
+### In [`src/compute-client/src`](../../../../src/compute-client/src/)
```

Comment:
Do absolute paths work in Github, i.e., could that be replaced with ``[`src/compute-client/src`](/src/compute-client/src/)``? It would make the links more robust in the face of reorganizations of the docs folder structure.

Reply:
Thanks, this seems to work! And from what I can see this style also links to the current branch in GitHub.